packages/core/src/sync-historical/service.ts

```diff
@@ -952,37 +953,40 @@ export class HistoricalSyncService extends Emittery<HistoricalSyncEvents> {
     fromBlock: Hex;
     toBlock: Hex;
   }): Promise<RpcLog[]> => {
+    const _params: ParseGetLogsErrorParameters["params"] = [
+      {
+        fromBlock: params.fromBlock,
+        toBlock: params.toBlock,
+
+        topics: params.topics,
+        address: params.address
+          ? Array.isArray(params.address)
+            ? params.address.map((a) => toLowerCase(a))
+            : toLowerCase(params.address)
+          : undefined,
+      },
+    ];
+
     try {
       return this.requestQueue.request({
         method: "eth_getLogs",
-        params: [
-          {
-            fromBlock: params.fromBlock,
-            toBlock: params.toBlock,
-
-            topics: params.topics,
-            address: params.address
-              ? Array.isArray(params.address)
-                ? params.address.map((a) => toLowerCase(a))
-                : toLowerCase(params.address)
-              : undefined,
-          },
-        ],
+        params: _params,
       });
     } catch (err) {
-      const retryRanges = getLogFilterRetryRanges(
-        err as LogFilterError,
-        params.fromBlock,
-        params.toBlock,
-      );
+      const getLogsErrorResponse = parseGetLogsError({
+        params: _params,
+        error: err as RpcError,
+      });
+
+      if (!getLogsErrorResponse.shouldRetry) throw err;
```

Comment:
Should we add a debug-level log here? It might be useful.

Reply:
Nice idea